### PR TITLE
Add categorical intializer for Tensor

### PR DESCRIPTION
**Pull request by xihui-wu**
This PR introduces capability of drawing samples from a categorical distribution into Tensor's intializer.

### CODE REVIEW

File: Sources/TensorFlow/Initializers.swift [hunk 1]
```diff
@@ -453,6 +453,31 @@ public extension Tensor where Scalar: TensorFlowFloatingPoint {
...
+    ///
+    /// - Attrs:
+    ///     - seed: If either seed or seed2 is set to be non-zero, the internal random number
+    ///         generator is seeded by the given seed.  Otherwise, a random seed is used.
+    ///     - seed2: A second seed to avoid seed collision.
+    ///
+    /// - Output: 2-D Tensor with shape `[batch_size, num_samples]`.  Each slice `[i, :]`
+    ///     contains the drawn class labels with range `[0, num_classes)`.
+    init<T: TensorFlowNumeric>(
+        logits: Tensor<T>,
```
Comment (marcrasi) [inline Sources/TensorFlow/Initializers.swift]: I suggest naming this label `randomCategorialLogits`, so that it describes what the initializer does with the logits.

File: Sources/TensorFlow/Initializers.swift [hunk 2]
```diff
@@ -453,6 +453,31 @@ public extension Tensor where Scalar: TensorFlowFloatingPoint {
...
+        numSamples: Tensor<Int32>,
```
Comment (marcrasi) [inline Sources/TensorFlow/Initializers.swift]: The standard in swift is to always use "count" instead of "num". I suggest `sampleCount`.

Also, since this is always an integer scalar, I suggest making it a more swift-native type like `Int`, and make the implementation convert it to the appropriate tensor type before passing it to the op.

File: Sources/TensorFlow/Initializers.swift [hunk 3]
```diff
@@ -453,6 +453,31 @@ public extension Tensor where Scalar: TensorFlowFloatingPoint {
...
+        seed: Int64 = 0,
```
Comment (marcrasi) [inline Sources/TensorFlow/Initializers.swift]: The other `random` initializers in this file use `TensorFlowSeed` with default value `Context.local.randomSeed`. This should do the same thing with this one, for consistency.

File: Sources/TensorFlow/Initializers.swift [hunk 4]
```diff
@@ -453,6 +453,31 @@ public extension Tensor where Scalar: TensorFlowFloatingPoint {
...
+        seed2: Int64 = 0
```
Comment (marcrasi) [inline Sources/TensorFlow/Initializers.swift]: Do you know why there are 2 seeds? Would it make sense to also make this one `seed2: TensorFlowSeed = Context.local.randomSeed`?

File: Sources/TensorFlow/Initializers.swift [hunk 5]
```diff
@@ -453,6 +453,31 @@ public extension Tensor where Scalar: TensorFlowFloatingPoint {
     }
 }
 
+public extension Tensor where Scalar: TensorFlowIndex {
+    /// Creates a tensor by drawing samples from a categorical distribution.
+    ///
+    /// - Parameters:
```
Comment (marcrasi) [inline Sources/TensorFlow/Initializers.swift]: This should use the same documentation comment formatting as the other doc comments in this repo. In particular:
* Put all the parameters under "Parameters". Swift doesn't have a concept of "Attrs".
* Name the return section "Returns" instead of "Output".
* Rename the `batch_size`, `num_classes`, `num_samples` to match swift naming conventions (camel case and "count" instead of "num").

File: Sources/TensorFlow/Initializers.swift (hunk 4 above)
Comment (xihui-wu) [reply Sources/TensorFlow/Initializers.swift]: "An second seed to avoid seed collision" as per the doc. But as we synced offline, going to pass in one TensorFlowSeed and call statelessMultinomial inside.

File: Tests/TensorFlowTests/InitializerTests.swift [hunk 6]
```diff
@@ -170,6 +170,17 @@ final class InitializerTests: XCTestCase {
...
 
+    func testCategoricalFromLogits() {
+        let probabilities = Tensor<Float>([[0.5, 0.3, 0.2], [0.6, 0.3, 0.1]])
+        let logits = log(probabilities)
+        let t = Tensor<Int32>(randomCategorialLogits: logits, sampleCount: 1)
+
+        XCTAssertEqual(TensorShape([2, 1]), t.shape)
+        // Test all elements are in range of [0, 3)
+        XCTAssertEqual(1, (Tensor<Int32>(t .>= Tensor<Int32>([[0], [0]])).product()).scalar!)
+        XCTAssertEqual(1, (Tensor<Int32>(t .< Tensor<Int32>([[3], [3]])).product()).scalar!)
```
Comment (marcrasi) [inline Tests/TensorFlowTests/InitializerTests.swift]: It would be a bit simpler to use `.all()` instead of `.product()`, like this: `XCTAssertTrue((t .>= Tensor<Int32>([[0], [0]])).all())`

File: Sources/TensorFlow/Initializers.swift (hunk 5 above)
Comment (marcrasi) [reply Sources/TensorFlow/Initializers.swift]: Oh sorry, didn't notice this earlier: The indentation should follow the conventions at https://google.github.io/swift/#parameter-returns-and-throws-tags
